src/contexts/magnificentContext.js

```diff
@@ -13,11 +14,13 @@ export const MagnificentProvider = ({ children }) => {
 	return (
 		<MagnificentContext.Provider value={value}>
 			<BrowserRouter>
-				<AuthProvider>
-					<ThemeProvider>
-						<LoaderProvider>{children}</LoaderProvider>
-					</ThemeProvider>
-				</AuthProvider>
+				<UserProvider>
+					<AuthProvider>
+						<ThemeProvider>
+							<LoaderProvider>{children}</LoaderProvider>
+						</ThemeProvider>
+					</AuthProvider>
+				</UserProvider>
```

Comment:
instead of using many contexts you could have made single context and used it

Reply:
It would increase complexity for maintaining the code and building logic.